src/coreclr/src/tools/r2rdump/R2RDump.cs

```diff
@@ -90,20 +91,16 @@ public MetadataReader FindAssembly(MetadataReader metadataReader, AssemblyRefere
 
         private static unsafe MetadataReader Open(string filename)
         {
-            byte[] Image = File.ReadAllBytes(filename);
+            byte[] image = File.ReadAllBytes(filename);
 
-            fixed (byte* p = Image)
-            {
-                IntPtr ptr = (IntPtr)p;
-                PEReader peReader = new PEReader(p, Image.Length);
-
-                if (!peReader.HasMetadata)
-                {
-                    throw new Exception($"ECMA metadata not found in file '{filename}'");
-                }
+            PEReader peReader = new PEReader(ImmutableArray.Create(image));
```

Comment:
Nit: You can use Unsafe.As to avoid creating a second copy of the whole array. Similar to what we do here: https://github.com/dotnet/runtime/blob/master/src/coreclr/src/tools/crossgen2/ILCompiler.Reflection.ReadyToRun/ReadyToRunReader.cs#L280

Reply:
Changed